### ISSUE TITLE
Inconsistent augmentations in the multi-target case

### ISSUE THREAD

**Issue by emirkonuk** [albumentations-team/albumentations#643]
## 🐛 Bug

Inconsistent augmentations in the multi-target case for HorizontalFlip(). Say we have an input_image and a target_image and we want to apply the same HorizontalFlip() consistently. Happens only when using with a pytorch Dataset object. 

## To Reproduce

```
import torch
import cv2
import albumentations
import numpy as np

transform = albumentations.Compose([albumentations.HorizontalFlip()],
                                   additional_targets={'target_image':'image'})
class Dataset(torch.utils.data.Dataset):
    def __init__(self, transform=None):
        super(Dataset, self).__init__()
        self.transform = transform
        self.length = 100
        self.input_image = np.random.rand(224,224,3)
        
        
    def __len__(self):
        return self.length 
        
    def __getitem__(self, idx):        
        augmented = self.transform(image=self.input_image, 
                                   target_image=self.input_image)
        input_image = augmented['image']
        target_image = augmented['target_image']
        return input_image, target_image
```

```
dataset = Dataset(transform)
for idx in range(100):
    assert np.alltrue(dataset[idx][1] == dataset[idx][0])
```

## Expected behavior

It works without a pytorch Dataset object:

```
for _ in range(100):
    input_image = np.random.rand(224,224,3)
    transform = albumentations.Compose([albumentations.HorizontalFlip()],
                                       additional_targets={'target_image':'image'})
    augmented = transform(image=input_image, target_image=input_image)
    assert np.alltrue(augmented['image'] == augmented['target_image'])
```

## Environment

 - Albumentations version: 0.4.5
 - Python version: 3.8.2
 - OS: Ubuntu 18.04
 - How you installed albumentations: conda

**Comment by suishicity** [comment]
I also met this problem, how do you solve it?

**Comment by emirkonuk** [comment]
> I also found this problem, how do you solve it?

Sadly, I didn't. I wrote my own augmentation functions instead.

**Comment by Dipet** [comment]
Sorry for the late replyю
You test has invalid behavior.
When you call `dataset[idx][1] == dataset[idx][0]` you has 2 different calls of aumgentation, so on augmentation has 2 different random seed on call.
Try this:
```python
dataset = Dataset(transform)
for idx in range(100):
    res = dataset[idx]
    assert np.alltrue(res[1] == res[0])
```
